worker-core/src/main/java/com/hpe/caf/worker/core/StreamingWorkerThreadPool.java

```diff
@@ -119,6 +122,17 @@ public int abortTasks() {
         public void afterExecute(Runnable r, Throwable t) {
             try {
                 super.afterExecute(r, t);
+                if (t == null && r instanceof Future<?>) {
+                    try {
+                        ((Future<?>) r).get();
+                    } catch (CancellationException ce) {
+                        t = ce;
```

Comment:
Seems like `CancellationException` should not trigger the worker to crash.  It looks like it would be triggered if there was a temporary disturbance in the connection to RabbitMQ.